File: server/src/services/tokenService.ts
/**
 * @fileoverview JWT session token utilities.
 * Room-scoped, anonymous JWTs. No user accounts.
 * @module services/tokenService
 */

import jwt from 'jsonwebtoken';
import { config } from '../config/env.js';
import { createLogger } from '../observability/logger.js';

const logger = createLogger('TokenService');

export interface SessionTokenPayload {
  userId: string;
  sessionId: string;
  roomCode: string;
  role: 'admin' | 'partner';
  nickname: string;
}

/**
 * Signs a room-scoped session JWT.
 */
export function signSessionToken(payload: SessionTokenPayload): string {
  return jwt.sign(payload, config.jwtSecret, {
    expiresIn: '24h' as unknown as number,
    issuer: 'cinepair',
  } as jwt.SignOptions);
}

/**
 * Verifies and decodes a session JWT.
 * Returns null if invalid/expired.
 */
export function verifySessionToken(token: string): SessionTokenPayload | null {
  try {
    const decoded = jwt.verify(token, config.jwtSecret, { issuer: 'cinepair' });
    return decoded as SessionTokenPayload;
  } catch (err) {
    logger.debug({ err }, 'Invalid session token');
    return null;
  }
}

File: backend/app/services/token_service.py
from app.core.security import TokenPayload, sign_session_token, verify_session_token


class TokenService:
    """Service for managing session tokens."""

    def create_token(
        self,
        user_id: str,
        session_id: str,
        room_code: str,
        role: str,
        nickname: str,
    ) -> str:
        """Create a new session token."""
        payload = TokenPayload(
            user_id=user_id,
            session_id=session_id,
            room_code=room_code,
            role=role,
            nickname=nickname,
        )
        return sign_session_token(payload)

    def verify_token(self, token: str) -> TokenPayload | None:
        """Verify and decode a session token."""
        return verify_session_token(token)
